Procurement/ViewModel/Recipes/VaalOrbRecipe.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using POEApi.Model;
+
+namespace Procurement.ViewModel.Recipes
+{
+    internal class VaalOrbRecipe : Recipe
+    {
+        private const int NeededVaalGems = 7;
+        public override string Name => "1 Vaal Orb";
+
+        public override IEnumerable<RecipeResult> Matches(IEnumerable<Item> items)
+        {
+            List<RecipeResult> recipeSets = new List<RecipeResult>();
+
+            var candidateGems = items.OfType<Gem>().Where(gem => gem.Corrupted).Cast<Item>().ToList();
+
+            var fragments = items.Where(x => x.TypeLine.StartsWith("Sacrifice at", StringComparison.CurrentCultureIgnoreCase)).ToList();
+            while(candidateGems.Count > 0)
+            {
+                var recipeResult = new RecipeResult()
+                {
+                    Instance = this,
+                    MatchedItems = new List<Item>(),
+                    Missing = new List<string>(),
+                    IsMatch = true
+                };
+
+                for (int i = candidateGems.Count - 1; i >= 0; i--)
+                {
+                    var candidateGem = candidateGems[i];
+                    if (recipeResult.MatchedItems.Count == NeededVaalGems)
+                    {
+                        break;
+                    }
+
+                    recipeResult.MatchedItems.Add(candidateGem);
+                    candidateGems.Remove(candidateGem);
+                }
+
+                var numberOfMissingGems = NeededVaalGems - recipeResult.MatchedItems.Count;
+
+                if (numberOfMissingGems == 1)
+                {
+                    recipeResult.Missing.Add($"{numberOfMissingGems} Vaal Skill gem");
```

Comment:
Maybe the compiler optimizes it away anyway, but since we know here `numberOfMissingGems` is 1, we can hard-code it into the string literal, which would be (perhaps trivially) faster.

Reply:
Good spot